roxie/ccd/ccdfile.cpp

```diff
@@ -240,7 +240,10 @@ class CRoxieLazyFileIO : implements ILazyFileIO, implements IDelayedFile, public
     IFileIO *getCheckOpen(unsigned &activeIdx)
     {
         CriticalBlock b(crit);
-        _checkOpen();
+        //Duplicate the check on current here so that the fast path (where it is already open) minimizes the time
```

Comment:
Could do the same in checkOpen() function above? Not sure how often that is used though.

This change looks ok as far as it goes though.